### PR TITLE
GAWB-1342: Create new method by 'duplicating'

### PR DESCRIPTION
**Pull request by MattPutnam**
Also quite a bit of code cleanup, converting the methods table to the new table, and redesigning the method detail view a bit to accommodate the new button.

The old table implementation had the option to add filter groups (in this case all/methods/configs).  In converting the methods table, I didn't add this to the new table component, but rather implemented it externally.  I'd like to eventually get rid of that design and do something like what the new workspace list does instead.  Keeping the filter groups external will facilitate/encourage this.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/common/components.cljs [hunk 1]
```diff
@@ -777,3 +777,41 @@
             :onClick #(swap! state assoc :collapsed? (not (:collapsed? @state)))})
           body]
          body)))})
+
+(react/defc FilterGroupBar
```
Comment (davidangb) [inline src/cljs/main/broadfcui/common/components.cljs]: could this `FilterGroupBar` be aligned with the FilterGroupBar in table_utils.cljs? There's a lot of duplicated code between the two.

**Review by davidangb — COMMENTED** [review]
One comment on code inline, not much else to say syntax-wise.

Regarding functionality:
1. Should we support duplicating configs as well as methods? Currently, we do ... is that correct?
2. This PR doesn't match exactly the details of the Jira ticket: "Copy of {foo}" vs. "{foo} copy"; "Upload" vs. "Create New Method". Could you please either update the PR or update the Jira ticket? Otherwise you'll get rejected during QA.
3. If I select "Duplicate method," fill in a namespace, then click "Upload" ... the copy is saved, but I remain on the original method. Looks like the jira ticket specifies I should be brought to my new copy instead.

**Comment by zarsky-broad** [comment]
There's a plan in the [mid-term] future to totally overhaul this table, so I wouldn't worry too much about the whole filter thing for now. Just FYI.

File: src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs [hunk 2]
```diff
@@ -45,32 +48,43 @@
...
-                                                                      :on-delete (:on-delete props)}])}]]
-        (clear-both)])
-     [:div {:style {:border style/standard-line
-                    :backgroundColor (:background-light style/colors)
-                    :borderRadius 8 :padding "1em" :marginTop "1em"}}
-      [:div {:style {:fontSize "120%" :marginBottom "0.5em"}}
-       (if workspace-id "Import as:" "Export to Workspace as:")]
-      (map
+       [:div {:style {:flex "0 0 290px" :paddingRight "1rem"}}
+        (when-not config?
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs]: right now, possible entities in the methods repo are "Workflow", "Task", and "Configuration". We're trying to do away with Tasks, but they still exist ... and this check allows duplication of both Tasks and Workflows. I think - and it's a PO call - we might only want to allow duplication on Workflows.

**Review by davidangb — COMMENTED** [review]
one last functional comment inline, looking good otherwise! If PO approves as-is - including duplication of Tasks - then 👍 . If PO doesn't want duplication of Tasks, then this needs a revision.

**Review by davidangb — APPROVED** [review]
👍 ! Didn't test the last two commits; just read the code (and it's small code)

**Pull request merged**